### PR TITLE
Avs post kyma provision

### PR DESCRIPTION
**Pull request by abbi-gaurav**

### CODE REVIEW

File: components/kyma-environment-broker/internal/avs/internal_eval_assistant.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+func NewInternalEvalAssistant(avsConfig Config) *InternalEvalAssistant {
+	return &InternalEvalAssistant{avsConfig: avsConfig}
+}
+
+func (iec *InternalEvalAssistant) CreateBasicEvaluationRequest(operations internal.ProvisioningOperation, url string) (*BasicEvaluationCreateRequest, error) {
+	return newBasicEvaluationCreateRequest(operations, iec, iec.avsConfig.GroupId, url)
+}
+
+func (iec *InternalEvalAssistant) SetOverrides(inputCreator internal.ProvisionInputCreator, evaluationId int64) {
+	inputCreator.SetOverrides("avs-bridge", []*gqlschema.ConfigEntryInput{
```
Comment (polskikiel) [inline components/kyma-environment-broker/internal/avs/internal_eval_assistant.go]: please use supported `AppendOverrides` function. `SetOverrides` has been deprecated, see [here](https://github.com/kyma-incubator/compass/blob/master/components/kyma-environment-broker/internal/model.go#L19).

Comment (polskikiel) [reply components/kyma-environment-broker/internal/avs/internal_eval_assistant.go]: then I would think to rename also `SetOverrides` function from `EvalAssistant` interface to `AppendOverrides` to be consistent

File: components/kyma-environment-broker/internal/avs/delegator.go [hunk 2]
```diff
@@ -34,15 +27,15 @@ func NewDelegator(avsConfig Config, operationsStorage storage.Operations) *Deleg
...
 }
 
-func (del *Delegator) DoRun(logger logrus.FieldLogger, operation internal.ProvisioningOperation, modelSupplier func(provisioningOperation internal.ProvisioningOperation) (*BasicEvaluationCreateRequest, error)) (internal.ProvisioningOperation, time.Duration, error) {
+func (del *Delegator) DoRun(logger logrus.FieldLogger, operation internal.ProvisioningOperation, evalAssistant EvalAssistant, url string) (internal.ProvisioningOperation, time.Duration, error) {
 	logger.Infof("starting the step")
 
-	if operation.AvsEvaluationInternalId != 0 {
+	if evalAssistant.CheckIfAlreadyDone(operation) {
 		msg := fmt.Sprintf("step has already been finished previously")
 		return del.operationManager.OperationSucceeded(operation, msg)
```
Comment (polskikiel) [inline components/kyma-environment-broker/internal/avs/delegator.go]: please change this line to:
```
return operation, 0, nil
```
to skip this step.

Currently the
```
OperationSucceeded(operation, msg)
```
ends the whole provisioning process with success

Comment (polskikiel) [reply components/kyma-environment-broker/internal/avs/delegator.go]: don't mind, we fix it in this PR: https://github.com/kyma-incubator/compass/pull/1019

**Comment by abbi-gaurav** [comment]
@polskikiel , I applied the feedback. please have a look.
I am waiting for Gabor to share the external tester access id so that I can test the post provisioning flow.

File: components/kyma-environment-broker/internal/avs/delegator.go (hunk 2 above)
Comment (polskikiel) [reply components/kyma-environment-broker/internal/avs/delegator.go]: please return `return operation, 0, nil` here

Comment (abbi-gaurav) [reply components/kyma-environment-broker/internal/avs/delegator.go]: ok, from the previous comment, it seems I do not need to change it.

**Pull request merged**